### PR TITLE
fix typing on set_cookie

### PR DESCRIPTION
**Pull request by FlorianLudwig**
## What do these changes do?

Change type hints on `set_cookie` to match the existing documentation

## Are there changes in behavior for the user?

For users that use linting, yes.

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by asvetlov** [comment]
Please fix CI, mypy linter is failed

**Comment by FlorianLudwig** [comment]
in my humble opinion that is a bug in typeshed and opened it there: https://github.com/python/typeshed/issues/3059

**Comment by asvetlov** [comment]
Thank you for the PR.
I think we should wait for upstream fix first

**Comment by FlorianLudwig** [comment]
The upstream bug is fixed in MyPy 0.720 as it pulls in typeched [f5c107cacd7c17ef541278bf228ca411fdeaa11b](https://github.com/python/typeshed/commits/f5c107cacd7c17ef541278bf228ca411fdeaa11b)

**Comment by webknjaz** [comment]
@FlorianLudwig the checks should pass before merging.

**Comment by FlorianLudwig** [comment]
@webknjaz Bumping mypy to 0.720 is needed - which results in other new linting errors (see travis).

I can check those later, maybe over the weekend

**Comment by asvetlov** [comment]
> I can check those later, maybe over the weekend

Please do

File: aiohttp/web_response.py [hunk 1]
```diff
@@ -170,8 +170,8 @@ def set_cookie(self, name: str, value: str, *,
                    max_age: Optional[Union[int, str]]=None,
                    path: str='/',
                    secure: Optional[str]=None,
```
Comment (serhiy-storchaka) [inline aiohttp/web_response.py:172]: What about `secure`?

File: aiohttp/web_response.py [hunk 2]
```diff
@@ -170,8 +170,8 @@ def set_cookie(self, name: str, value: str, *,
...
-                   httponly: Optional[str]=None,
-                   version: Optional[str]=None) -> None:
+                   httponly: Optional[bool]=None,
+                   version: Optional[bool]=None) -> None:
```
Comment (serhiy-storchaka) [inline aiohttp/web_response.py:174]: Is `version` really a boolean?

**Comment by serhiy-storchaka** [comment]
And, if the current tests are passed with updated MyPy, would be nice to add tests for `set_cookie()` with these parameters.

**Comment by asvetlov** [comment]
The problem is fixed by e4573f8c8 and 1220613b5

**Pull request closed without merging**